src/charm.py

```diff
@@ -483,12 +484,15 @@ def _workload_reset(self) -> StatusBase:
             A `StatusBase` to be set by the caller
         """
         try:
+            primary_address = self._get_primary_address_from_peers()
+            if not primary_address:
+                logger.debug("Primary not yet defined on peers. Skipping workload reset")
+                return MaintenanceStatus("Workload reset")
```

Comment:
aren't these a little conflicting messages? 
Logging that you're skipping reseting the workload - but. then setting maintaince message that you are resetting the workload